tests/target/issue_1255.rs

```diff
@@ -1,8 +0,0 @@
-#![feature(specialization)]
-
-trait Trait {
-    type Type;
-}
-impl<T> Trait for T {
-    default type Type = u64;
-}
```

Comment:
Oh, I see you've made that change here.

So, when structuring the commits in the PR, it is usually best to fix them all up at the end so that they are clearer for a reviewer or person reading commits in the tree, rather than reflecting the evolution of the code as it was written. So, for this PR I would have one commit for the test (squashing in the test changes), one for the changes (squashing the refactoring into the first change), and one for reformatting.